operator/pkg/component/component/component.go

```diff
@@ -626,22 +647,27 @@ func (c *IngressComponent) Namespace() string {
 	return c.CommonComponentFields.Namespace
 }
 
+// Enabled implements the IstioComponent interface.
+func (c *IngressComponent) Enabled() bool {
+	// type assert is guaranteed to work in this context.
+	return boolValue(c.spec.(*v1alpha1.GatewaySpec).Enabled)
```

Comment:
I know this is a back-port, but this is really bad. Operator already has ~50k lines of code that mostly do some of struct/json/yaml mangling, and here we go and re-invent the wheel once again. Enablement of all components can be checked via tpath in a generic fashion, why add boilerplate functions like this one?

Reply:
I get your point and I agree for cases where there's no component specific logic we can drop the interface or pass it through to a general function (like we do in renderManifest). The tradeoff is that for component specific code, things can become less readable if we have a switch with a bunch of case statements. 
This code actually started as just such a function but the selection logic was brittle and hard to understand (type asserts) and even though the interface approach introduces some pass through wrapper functions I think overall it's more readable. At least that's how it felt when I made the change!
But in general I'm glad to hear someone complaining about the state of code quality because it means you care and there are definitely a *ton* of opportunities to improve and prune the code. I've done a bit already here and there but there hasn't been much time. One of the top goals in 1.6 will be cleanup and pruning and I hope you and others will contribute ideas and PRs on how to improve the code base. We can start off by creating a list of offenders, discuss and tackle it as a work item in the weekly sync.

Reply:
That sounds very good to me. Thank you for not taking my rather snarky comment personally :) I'll be happy to try and come up with some ideas.